src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource/DiagnosticSourceEventSourceListener.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace System.Diagnostics;
+
+internal sealed class DiagnosticSourceEventSourceListener : IDisposable
+{
+    private const string c_ActivitySourcePrefix = "[AS]";
+    private const string c_ParentRatioSamplerPrefix = "ParentRatioSampler(";
+
+    private DiagnosticSourceEventSourceFilterAndTransform? _specList;
+    private DiagnosticSourceEventSourceActivitySourceListener? _activitySourceListener;
```

Comment:
To reduce the number of abstractions I would just inline the ActivitySourceListener into this class. Both of them are pretty small, this type is already doing all the parsing for Activity specs and there is no scenario I can see where we'd use the ActivitySourceListener without also needing an instance of this type wrapped around it.